Sources/MessagingPush/PushClickHandler.swift

```diff
@@ -118,7 +120,11 @@ class PushClickHandlerImpl: NSObject, PushClickHandler, UNUserNotificationCenter
     }
 
     private func pushClicked(_ response: UNNotificationResponse, parsedPush: CustomerIOParsedPushPayload) {
-        // TODO: prevent duplicate push metrics and deep link handling.
+        guard !pushHistory.hasHandledPushClick(deliveryId: parsedPush.deliveryId) else {
+            // push has already been handled. exit early
+            return
+        }
+        pushHistory.handledPushClick(deliveryId: parsedPush.deliveryId)
```

Comment:
Did we write any tests for `pushClicked` anywhere? If not, can we write some?

Reply:
This feature has been heavily reliant on QA testing thus far. Automated tests around swizzling and UserNotifications has been difficult to write. I have not yet found an easy or quick way to make them. 

I do agree and think we can write some for this function after QA concludes, yes.

Reply:
I am going to merge this PR to make maintenance and QA testing easier. I made a note to add these tests and codecov in the target PR should help to keep track of these missing tests.